src/plugins/SoundPlugin.js

```diff
@@ -0,0 +1,385 @@
+  onMusicToggle() {
+    this.musicMuted = !this.musicMuted;
+    this._checkSoundMute();
+  }
+
+  /**
+   * @memberof SoundPlugin
+   */
+  onVOToggle() {
+    this.voMuted = !this.voMuted;
+    this._checkSoundMute();
+  }
+
+  /**
+   * @memberof SoundPlugin
+   */
+  onSFXToggle() {
+    this.sfxMuted = !this.sfxMuted;
+    this._checkSoundMute();
+  }
+
+  /**
+   * @memberof SoundPlugin
+   */
+  _checkSoundMute() {
+    this.soundMuted = this.sfxMuted && this.voMuted && this.musicMuted;
+  }
+
+  /**
+   *
+   *
+   * @param {*} key
+   * @param {*} value
+   * @param {*} element
+   * @memberof SoundPlugin
+   */
+  setMuteProp(key, value, element) {
+    this[key] = value;
+    this._setMuteProp(key, element, value);
+  }
+
+  /**
+   *
+   *
+   * @memberof SoundPlugin
+   */
+  opened() {
+    if (this.soundButton !== null) {
+      this.soundButton.classList.remove('disabled');
+    }
+
+    if (this.sfxButton !== null) {
+      this.sfxButton.classList.remove('disabled');
+    }
+
+    if (this.voButton !== null) {
+      this.voButton.classList.remove('disabled');
+    }
+
+    if (this.musicButton !== null) {
+      this.musicButton.classList.remove('disabled');
+    }
+
+    this.soundMuted = !!SavedData.read('soundMuted');
+    this.musicMuted = !!SavedData.read('musicMuted');
+    this.sfxMuted = !!SavedData.read('sfxMuted');
+    this.voMuted = !!SavedData.read('voMuted');
+  }
+
+  /**
+   * @memberof SoundPlugin
+   */
+  close() {
+    this._disableButton(this.soundButton);
+    this._disableButton(this.voButton);
+    this._disableButton(this.sfxButton);
+    this._disableButton(this.musicButton);
+  }
+
+  /**
+   * @memberof SoundPlugin
+   */
+  teardown() {
+    if (this.soundButton !== null) {
+      this.soundButton.removeEventListener(
+        'click',
+        this.onSoundToggle.bind(this)
+      );
+    }
+
+    if (this.musicButton !== null) {
+      this.musicButton.removeEventListener(
+        'click',
+        this.onMusicToggle.bind(this)
+      );
+    }
+
+    if (this.sfxButton !== null) {
+      this.sfxButton.removeEventListener('click', this.onSFXToggle.bind(this));
+    }
+
+    if (this.voButton !== null) {
+      this.voButton.removeEventListener('click', this.onVOToggle.bind(this));
+    }
+  }
+
+  /**
+   *
+   *
+   * @param {HTMLInputElement} slider
+   * @param {string} elementName
+   * @memberof SoundPlugin
+   */
+  sliderSetup(slider, elementName) {
+    if ('range' !== slider.type) {
+      console.warn(
+        `SpringRoll Container: Sound plugin was passed a invalid input of ${elementName}. Input must be of type range`
+      );
+      return null;
+    }
+
+    slider.min = '0';
+    slider.max = '1';
+    slider.step = '0.1';
+    return slider;
+  }
+
+  /**
+   * Controls the volume range
+   * @param {number} i
+   * @param {number} [min=0]
+   * @param {number} [max=1]
+   * @returns
+   * @memberof SoundPlugin
+   */
+  volumeRange(i, min = 0, max = 1) {
+    switch (true) {
```

Comment:
fun little trick I learned about switch case statements. If the switch value is true, it will run whatever case is returns true. Great for when you have a lot of if statements